src/libmongoc/CMakeLists.txt

```diff
@@ -99,6 +99,12 @@ if (NOT ENABLE_ZLIB STREQUAL "OFF")
    include (CheckIncludeFiles)
    check_include_files ("unistd.h" HAVE_UNISTD_H)
    check_include_files ("stdarg.h" HAVE_STDARG_H)
+   if (HAVE_UNISTD_H)
+      add_definitions (-DHAVE_UNISTD_H)
+   endif ()
+   if (HAVE_STDARG_H)
+      add_definitions (-DHAVE_STDARG_H)
+   endif ()
```

Comment:
Offhand, PHP was [already defining `HAVE_UNISTD_H` for bundled zlib](https://github.com/mongodb/mongo-php-driver/blob/1.12.0/config.m4#L434). It was trivial to make the change for `HAVE_STDARG_H`, although I didn't notice any build failure omitting it on Ubuntu 20.04.

LGTM!

Reply:
(Lack of) `<stdarg.h>` was not causing issues, as far as I could observe. `HAVE_STDARG_H` was just added to compliment the addition of `HAVE_UNISTD_H` for consistency. The (lack of) `<unistd.h>` was the relevant issue being addressed.